src/gsy_e/models/strategy/__init__.py

```diff
@@ -669,6 +674,7 @@ def update_offer_rates(self, market: "OneSidedMarket", updated_rate: float) -> N
                 continue
             try:
                 # Delete the old offer and create a new equivalent one with an updated price
+                time_slot = offer.time_slot or iterated_market.time_slot
```

Comment:
shouldn't they both be the same?

Reply:
For the future market not really, because iterated_market.time_slot is None, and offer.time_slot is the actual future time_slot that the offer has been placed to.

Reply:
Ah you're right, I always miss that
Thanks!